src/Umbraco.Web.UI.Client/src/views/common/infiniteeditors/iconpicker/iconpicker.controller.js

```diff
@@ -43,10 +43,24 @@ function IconPickerController($scope, localizationService, iconHelper) {
 
         setTitle();
 
-        iconHelper.getIcons().then(function (icons) {
-            vm.icons = icons;
-            vm.loading = false;
-        });
+        iconHelper.getAllIcons()
+            .then(icons => {
+                vm.icons = icons;
+
+                // Get's legacy icons, removes duplicates then maps them to IconModel
+                iconHelper.getIcons()
+                    .then(icons => {
+                        if (icons && icons.length > 0) {
+                            let legacyIcons = icons
+                                .filter(icon => !vm.icons.find(x => x.name == icon))
+                                .map(icon => { return { name: icon, svgString: undefined }; });
```

Comment:
I would use `null` instead of `undefined` here since you actually assign a value to the property.

Reply:
Agreed, makes a lot of sense 👍